synapse/handlers/message.py

```diff
@@ -795,6 +795,23 @@ async def create_new_client_event(
         if requester:
             context.app_service = requester.app_service
 
+        third_party_result = await self.third_party_event_rules.check_event_allowed(
+            event, context
+        )
+        if not third_party_result:
```

Comment:
I suspect it is fine, but returning an empty dict (`{}`) will result in the event being rejected. I don't think that's even a sane thing to return though so it is probably fine?

Reply:
Yes, I think that's fine.